### DISCUSSION TITLE
Automatic authorize! to all action

### DISCUSSION THREAD

**Discussion by ghost** [palkan/action_policy#196] [Q&A]
Hello, is there a way to authorize! all action of all controllers without having to write it everywhere ? Thanks.

**Comment by palkan** [comment]
Hi!
Do you mean something like this: https://actionpolicy.evilmartians.io/#/rails?id=verify_authorized-hooks ?

**Comment by coder2000** [comment]
I know this is an older question but would something like:
```rb
after_action :authorize!
```
work as expected?

**Reply by palkan** [reply]
It would behave differently:
- `verify_authorized` doesn't perform any authorizations, it just checks that at least once has been performed during the action execution
- `after_action :authorize!` would try to perform authorization after the action; first, it's conceptually incorrect, since we want to authorize before the action has been performed; secondly, it would only work with implicit targets.

**Reply by coder2000** [reply]
I understand that, I was trying to answer the original question where he didn't want to have to use `authorize!` in every action.
